src/attributes.jl

```diff
@@ -394,13 +402,39 @@ function submit(model::ModelLike, sub::AbstractSubmittable, args...)
     end
 end
 
+## Submittables
+
 """
-    SettingSingleVariableFunctionNotAllowed()
+    LazyConstraint(callback_data)
 
-Error type that should be thrown when the user [`set`](@ref) the
-[`ConstraintFunction`](@ref) of a [`SingleVariable`](@ref) constraint.
+Lazy constraint `func`-in-`set` submitted as a tuple `(func, set)`. The optimal
+solution returned by [`VariablePrimal`](@ref) will satisfy all lazy
+constraints that have been submitted.
+
+This can be submitted only from the lazy callback (that is, the function set to
+[`LazyCallback`](@ref)). The field `callback_data` is a solver-specific
+callback type that is passed as the argument to the lazy callback.
 """
-struct SettingSingleVariableFunctionNotAllowed <: Exception end
+struct LazyConstraint <: AbstractSubmittable
+    callback_data::Any
+end
+
+"""
+    HeuristicSolution(callback_data)
+
+Heuristically obtained integer-feasible solutions. The solution is given by a
+`Dict{VariableIndex, Float64}` mapping the variables to their solution value.
```

Comment:
Why a `Dict{VariableIndex, Float64}`?

Wouldn't it be easier to pass a `Vector{VariableIndex}` containing the list of variables, and a `Vector{<:Number}` containing the associated values?

Then, to post a heuristic solution, one does something like
```julia
MOI.submit(model, ::MOI.CbHeuristicSolution, vars::Vector{VariableIndex}, vals::Vector{<:Number})
```

(I wrote `<:Number` to allow more multiple-precision and possibly complex numbers)

Reply:
Indeed, it would be more consistent with the vectorized version of `set`.